flow/ThreadHelper.actor.h

```diff
@@ -63,11 +72,27 @@ void doOnMainThreadVoid(Future<Void> signal, F f, Error* err) {
 // WARNING: The error returned in `err` can only be read on the FDB network thread because there is no way to
 // order the write to `err` with actions on other threads.
 //
+// WARNING: The Error member of `T` is accepted as a pointer to a data member so the caller can avoid dereferencing
+// `T` until it is initialized on the main thread.
+//
 // `onMainThreadVoid` is defined here because of the dependency in `ThreadSingleAssignmentVarBase`.
+// template <class F>
+// void onMainThreadVoid(F f, Error* err = nullptr, TaskPriority taskID = TaskPriority::DefaultOnMainThread) {
+// 	Promise<Void> signal;
+// 	internal_thread_helper::doOnMainThreadVoid(signal.getFuture(), f, err);
+// 	g_network->onMainThread(std::move(signal), taskID);
+// }
+template <class F, class T>
+void onMainThreadVoid(F f, T* t, Error T::*member, TaskPriority taskID = TaskPriority::DefaultOnMainThread) {
+	Promise<Void> signal;
+	internal_thread_helper::doOnMainThreadVoid(signal.getFuture(), f, t, member);
+	g_network->onMainThread(std::move(signal), taskID);
+}
+
 template <class F>
-void onMainThreadVoid(F f, Error* err = nullptr, TaskPriority taskID = TaskPriority::DefaultOnMainThread) {
+void onMainThreadVoid(F f, std::nullptr_t = nullptr, TaskPriority taskID = TaskPriority::DefaultOnMainThread) {
```

Comment:
Can we remove the second argument from this function?

Reply:
Yes I think that's fine, there are just a bunch of places that pass `nullptr` and I didn't want to change all of them initially :D (see https://github.com/apple/foundationdb/blob/34e6a8f94200f3738e8ee6fc58722a11fcbeb18e/fdbclient/AsyncTaskThread.h#L89 for example). I can clean these up now.